internal/services/domains/client/client.go

```diff
@@ -2,19 +2,23 @@ package client
 
 import (
 	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/manicminer/hamilton/msgraph"
 
 	"github.com/hashicorp/terraform-provider-azuread/internal/common"
 )
 
 type Client struct {
 	AadClient *graphrbac.DomainsClient
+	MsClient  *msgraph.DomainsClient
 }
 
 func NewClient(o *common.ClientOptions) *Client {
 	aadClient := graphrbac.NewDomainsClientWithBaseURI(o.AadGraphEndpoint, o.TenantID)
-	o.ConfigureClient(&aadClient.Client, o.AadGraphAuthorizer)
+	msClient := msgraph.NewDomainsClient(o.TenantID)
```

Comment:
I ended up setting this [in the ClientOptions.ConfigureClient() method](https://github.com/hashicorp/terraform-provider-azuread/blob/5d84cf4d72d6791ae0a8ca2dfa2c7917cb083896/internal/common/client_options.go#L34) rather than expand all the constructors to support args that would invariably have the same values for most users